### PR TITLE
plots: replace --show-json with --show-vega

### LINKED ISSUE #3890: plots: replace --show-json with --show-vega
Requested by @dmpetrov for cml. `--show-vega` should require a target and return a filled vega template. `--show-json` is not needed, let's delete it.

### PR DESCRIPTION
**Pull request by efiop**
Fixes #3890

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Are docs updated in an open PR somewhere? Or should I change it in one of mine?

File: dvc/command/plots.py [hunk 1]
```diff
@@ -138,10 +147,10 @@ def add_parser(subparsers, parent_parser):
         help="Required when CSV or TSV datafile does not have a header.",
     )
     plots_show_parser.add_argument(
-        "--show-json",
+        "--show-vega",
         action="store_true",
         default=False,
-        help="Show output in JSON format.",
+        help="Show output in VEGA format.",
```
Comment (jorgeorpinel) [inline dvc/command/plots.py:153]:
```suggestion
        help="Show output in Vega format.",
```

Comment (jorgeorpinel) [reply dvc/command/plots.py:153]: Same in line 216

**Comment by efiop** [comment]
Thank you @jorgeorpinel !

And I also forgot to update completion scripts :man_facepalming: Will do.

File: dvc/command/plots.py (hunk 1 above)
Comment (efiop) [reply dvc/command/plots.py:153]: Actually, these are fixed. Probably did that after.

Comment (jorgeorpinel) [reply dvc/command/plots.py:153]: I'm seeing this in the output of `dvc plots show -h` in `1.0.0a7+12172f` which I think is the latest `master` version.

Comment (efiop) [reply dvc/command/plots.py:153]: @jorgeorpinel Oh, sorry, I'm blind. I looked at the bottom part that is the patch itself and not your suggestion. Will fix, thank you!

**Pull request merged**